core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -362,6 +367,7 @@ enum SharePartitionState {
         this.leaderEpoch = leaderEpoch;
         this.maxInFlightRecords = maxInFlightRecords;
         this.maxDeliveryCount = maxDeliveryCount;
+        this.throttleRecordsDeliveryLimit = Math.max(2, (int) Math.ceil((double) maxDeliveryCount / 2));
```

Comment:
Define this as a constant and use elsewhere `MINIMUM_THROTTLE_RECORDS_DELIVERY_LIMIT`.
```suggestion
        this.throttleRecordsDeliveryLimit = Math.max(MINIMUM_THROTTLE_RECORDS_DELIVERY_LIMIT, (int) Math.ceil((double) maxDeliveryCount / 2));
```